cloud: use uuid as bucket name

jeffswenson opened the pull request:
Previously, the GCS and S3 tests used hard coded bucket names to test
the behavior of buckets that do not exist. Now, the tests use a random
UUID. This change is necessary because someone appears to have created
the GCP bucket which causes the test to fail.

Release Justification: Test Only Change
Release Note: None
Fixes: https://github.com/cockroachdb/cockroach/issues/135307
Fixes: https://github.com/cockroachdb/cockroach/issues/135348
Fixes: https://github.com/cockroachdb/cockroach/issues/135349
Fixes: https://github.com/cockroachdb/cockroach/issues/135350
Fixes: https://github.com/cockroachdb/cockroach/issues/135351
Fixes: https://github.com/cockroachdb/cockroach/issues/135352

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/135595)
<!-- Reviewable:end -->

jeffswenson committed:
cloud: use uuid as bucket name

Previously, the GCS and S3 tests used hard coded bucket names to test
the behavior of buckets that do not exist. Now, the tests use a random
UUID. This change is necessary because someone appears to have created
the GCP bucket which causes the test to fail.

Release Justification: Test Only Change
Release Note: None
Fixes: #135307
Fixes: #135348
Fixes: #135349
Fixes: #135350
Fixes: #135351
Fixes: #135352

jeffswenson commented:
This PR depends on #135619. The roll back and the fix need to be in two different PRs so the fix applies cleanly in all backports.

msbutler reviewed (approved):
might be best to kick off another cloud nightly

jeffswenson commented:
https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/17826078

jeffswenson commented:
The cloud azure test is failing, but that seems to be independent of this change.

https://github.com/cockroachdb/cockroach/issues/135661

jeffswenson commented:
Thanks for the review!
bors r+
